### ISSUE TITLE
Code Navigation settings for Starlark files

### ISSUE THREAD

**Issue by arquer** [bazel-contrib/vscode-bazel#414]
Probably not an actual issue but more like a question/feature request.

While browsing BUILD files, is there any way that I can configure where the extension will search for rules so that I can easily navigate them (Ctrl + Click)?

I ask because I have projects where my checked out directory structure might not match the one assumed by the rules in BUILD, and even when doing that I often find code navigation taking be to BUILD files located in my bazel cache or in a temporary "bazel-bin" or similar.

In other words, is there a way I can configure this extension so that Ctrl+Click on a dependency that points to another repo (@repo/path/to:rule) actually takes me to a cloned version of that repo I have?

Thanks!

**Comment by vogelsgesang** [comment]
This functionality should be available if you configure a language server (see the README). Currently, that functionality is still experimental, also see #1 for a lot more context on the current state of language server support

**Comment by arquer** [comment]
Oh I see, thanks, I didn't know a language server had anything to do with this so I hadn't looked into that, definitely will now.

**Comment by arquer** [comment]
Hi @vogelsgesang , I went ahead and installed the language server (bazel-lsp) but have seen no difference in the behaviors I was explaining above.
Code navigation often still takes me to bazel's cache instead of the starlark BUILD sources, even when I add bazel's cache to the list of directories to ignore.

Is there some way of debugging this you would recommend? I am pretty new to starlark/bazel and really all I want is code navigation so I can navigate a huge source code base easily.

**Comment by vogelsgesang** [comment]
I am a bit confused.

Which language are you trying to use code navigation on? Starlark? C++? Python? some other language?

**Comment by arquer** [comment]
Starlark! basically I wanna be able to ctrl + click on some dependencie of a bazel rule and be taken to that rule's definition

**Comment by cameron-martin** [comment]
It sounds like the ask here is for code navigation to resolve symlinks, like clangd does from memory. This may have unintended consequences on further code navigation when naively implemented, since it'll lose the context on what the bazel workspace is.

Do you mind opening an issue on bazel-lsp and starpls?

**Comment by cameron-martin** [comment]
> I believe this is a boundary case that is not worth handling separately, as for your usecase you could also work on the symlinked files.

I do think resolving symlinks when going to definitions is worth considering, since I have found this feature in other language servers (I believe clangd) useful in the past. However, I believe this lives in the language servers, not this extension.
